locust/contrib/geventhttpclient.py

```diff
@@ -61,23 +66,31 @@ class GeventHttpSession(object):
     def __init__(self, base_url):
         self.base_url = base_url
         self.cookiejar = CookieJar()
-        self.client = UserAgent(max_retries=1, cookiejar=self.cookiejar)
-        
+        self.client = LocustUserAgent(max_retries=1, cookiejar=self.cookiejar)
+    
+    def _build_url(self, path):
+        """ prepend url with hostname unless it's already an absolute URL """
+        if absolute_http_url_regexp.match(path):
+            return path
+        else:
+            return "%s%s" % (self.base_url, path)
+    
     def request(self, method, path, name=None, **kwargs):
-        url = self.base_url + path
+        # prepend url with hostname unless it's already an absolute URL
+        url = self._build_url(path)
```

Comment:
- you can probably simplify or remove the `_build_url` method and just use something like:

`url = path if path.startswith('http') else urljoin(base_url, path)`

Reply:
Hmm, changing to just check `path.startswith('http')` and using `urljoin()` would create a situation where this would works:

`self.client.get("some-path") # <-- no slash in beginning`

while this would not work:

`self.client.get("http-protocol")`

Maybe just switch to `urljoin()` but stick with the regexp to check if a full URL was specified?

Reply:
you could try parsing the string with `urllib.parse.urlparse()` ... but regex works too

Reply:
I opted to instead validate that the `host` doesn't contain a trailing slash, and just use string concatination, because `urljoin()` seems a bit overkill for this use-case since it parses the whole URL. Maybe it's premature optimization, but for this specific case I think it might not be.

```python
>>> from timeit import timeit
>>> timeit("url = urljoin(base, path)", setup='base="http://127.0.0.1"; path="/test"; from urllib.parse import urljoin;', number=100000)
1.4690891270292923
>>> timeit("url = base + path", setup='base="http://127.0.0.1"; path="/test";', number=100000)
0.012773031019605696
```